### ISSUE TITLE
Postal Code checks

### ISSUE THREAD

**Issue by yannickvandendijck** [influenzanet/study-manager-app#7]
Can we check where the postal code is a true postal code.
For example, by looking whether it is present in the list of all postal code of the country (eg, Belgium)?

**Comment by phev8** [comment]
Currently this question type is missing, but we have it on the todo list.

The idea is to have a text input where you can start to type then a list of possibilities appear. 
To prevent too much data usage by downloading the full list of postal codes every time when someone open the survey, it probably makes sense to have a server endpoint for validating entries. 

@cturbelin This was also a question type you mentioned us before, right?

**Comment by cturbelin** [comment]
Yes !
We will need this kind of lookup question : We store administrative municipality code (not known from participants) but ask for postal code (everybody knowns) and ask the participant to select the municipality in this postal code. Postal code is not used because it's overlapping on many administratives area and we dont have proper reference data from them (like demographic data).
We have implemented this kind of component in the old platform.

So it's a lookup, we can imagine an abstract component like :
- Get a list of responses from an input data on a predefined source (url or internal data), each response with code and label field
- If single response is provided store the provided response code (can be different from the input, this allow mapping or normalization)
- if several show response labels to be selected and store the response code

In our case we have a mapping table with over than 50,000 entries so it's not reasonable to use local storage but I think it can be useful in some other cases.

Maybe this kind of component can be used to check (single response from provider) or map with user selection to get the right response (or optionaly to force selection regardless the number of response if we need user validation)

**Comment by yannickvandendijck** [comment]
For Belgium, all the postal code are unique. So that helps.
For Belgium there are just under 3000 postal codes. So we need to check whether the postal code response is within the list of these 3000 postal codes. If it is not, we should have an error message.

**Comment by cturbelin** [comment]
Yes ! It seem to be the simplest use case for a lookup (check an entry is in the known entries). 
I think it's important that the lookup response can provide also a label (sometimes not the same as the entered code) so the user can check there is no mistyping. For postal code it can be the municipality, district..
